service/health/response.go

```diff
@@ -14,26 +14,40 @@ type ClusterStatus struct {
 	State  key.LifecycleState `json:"state"`
 }
 
-// NodeStatus holds data about the health/status of a node.
-type NodeStatus struct {
-	// Identity
+// NodeStatusComputeResources holds data about available or requested compute resources.
+type NodeStatusComputeResources struct {
+	CPU         int64 `json:"cpu"`
+	MemoryBytes int64 `json:"memory_bytes"`
+}
+
+// NodeStatusMachineResources holds data about the status of storage on a cluster node.
+type NodeStatusMachineResources struct {
+	CPUCount               int64 `json:"cpu_count"`
+	MemoryCapacityBytes    int64 `json:"memory_capacity_bytes"`
+	MemoryAllocatableBytes int64 `json:"memory_allocatable_bytes"`
+	EphemeralStorageCap    int64 `json:"ephemeral_storage_capacity_bytes"`
+	EphemeralStorageAvail  int64 `json:"ephemeral_storage_allocatable_bytes"`
+}
+
+// NodeStatusIdentity holds data about the identity of a node which is generally static.
+type NodeStatusIdentity struct {
 	Name               string `json:"name"`
+	KubeletVersion     string `json:"kubelet_version"`
+	OperatorVersion    string `json:"operator_version"`
 	Role               string `json:"role"`
 	IP                 string `json:"ip"`
 	Hostname           string `json:"hostname"`
 	InstanceType       string `json:"instance_type"`
 	AvailabilityRegion string `json:"availability_region"`
 	AvailabilityZone   string `json:"availability_zone"`
-	// Versions
-	KubeletVersion  string `json:"kubelet_version"`
-	OperatorVersion string `json:"operator_version"`
-	// Resources
-	CPUCount               int64 `json:"cpu_count"`
-	MemoryCapacityBytes    int64 `json:"memory_capacity_bytes"`
-	MemoryAllocatableBytes int64 `json:"memory_allocatable_bytes"`
-	EphemeralStorageCap    int64 `json:"ephemeral_storage_capacity_bytes"`
-	EphemeralStorageAvail  int64 `json:"ephemeral_storage_allocatable_bytes"`
-	// Status
-	Ready  bool       `json:"ready"`
-	Health key.Health `json:"health"`
+}
+
+// NodeStatus represents the health status of a single node in a cluster.
+type NodeStatus struct {
+	Ready            bool                       `json:"ready"`
+	Health           key.Health                 `json:"health"`
+	Identity         NodeStatusIdentity         `json:"identity"`
+	MachineResources NodeStatusMachineResources `json:"machine"`
+	LimitTotals      NodeStatusComputeResources `json:"limit_totals"`
+	RequestTotals    NodeStatusComputeResources `json:"request_totals"`
 }
```

Comment:
Do we have spec for those types written down somewhere?

Reply:
@cornelius-keller is working on a formal spec. For now, I am basing my work off of [our metrics spreadsheet](https://docs.google.com/spreadsheets/d/1nx2VdCwDEP2e5DlobjtFr7ANZTDso6_xDvzrs6sJu0A/edit#gid=0) and [this WIP docs PR](https://github.com/giantswarm/docs/pull/344). Things are in flux because we're not totally sure where this code is going to live: in this microservice, in an existing operator, in a new operator, or in the API.